File: resources/adapter/Javascript/Log.js
if (typeof ALittle == "undefined") ALittle = {};

// 设置打印的回调函数
ALittle.log = undefined;
function SetLogCallback(func)
{
    ALittle.log = func;
}

// 日志级别
ALittle.LogLevel =
{
    INFO : 0,
    WARN : 1,
    ERROR : 2,
}

// 日志函数
ALittle.Log = function()
{
    if (log == undefined)
        console.log(new_arg.join("\t"));
    else
        log(new_arg.join("\t"), 0);
}

// 日志函数
ALittle.Warn = function()
{
   if (log == undefined)
       console.warn(new_arg.join("\t"));
   else
       log(new_arg.join("\t"), 1);
}


// 日志函数
ALittle.Error = function()
{
   if (log == undefined)
       console.error(new_arg.join("\t"));
   else
       log(new_arg.join("\t"), 2);
}

File: resources/adapter/Base/Log.lua

module("ALittle", package.seeall)

local ___pairs = pairs
local ___ipairs = ipairs
local ___coroutine = coroutine

local concat = table.concat
local select = select
local tostring = tostring
LogLevel = {
	INFO = 0,
	WARN = 1,
	ERROR = 2,
}

function Log(...)
	local new_arg = {}
	local arg_count = select("#", ...)
	for i = 1, arg_count, 1 do
		Push(new_arg, tostring(select(i, ...)))
	end
	A_ScriptSystem:Log(concat(new_arg, "\t"), 0)
end

function Warn(...)
	local new_arg = {}
	local arg_count = select("#", ...)
	for i = 1, arg_count, 1 do
		Push(new_arg, tostring(select(i, ...)))
	end
	A_ScriptSystem:Log(concat(new_arg, "\t"), 1)
end

function Error(...)
	local new_arg = {}
	local arg_count = select("#", ...)
	for i = 1, arg_count, 1 do
		Push(new_arg, tostring(select(i, ...)))
	end
	A_ScriptSystem:Log(concat(new_arg, "\t"), 2)
end

